spec/services/scishield_api_adapter_spec.rb

```diff
@@ -4,83 +4,98 @@
+        adapter.certified? instance_double(ResearchSafetyCertificate, name: "RANDOM")
+      end
     end
-  end
 
-  describe "unable to connect" do
-    before do
-      stub_request(:get, api_endpoint)
-        .to_timeout
-    end
+    describe "unable to connect" do
+      before do
+        stub_request(:get, api_endpoint)
+          .to_timeout
+      end
 
-    it "raises an error" do
-      expect { adapter.certified?(anything) }.to raise_error(Timeout::Error)
+      it "raises an error" do
+        expect { adapter.certified?(anything) }.to raise_error(Timeout::Error)
+      end
     end
-  end
 
-  describe "the user is not found" do
-    let(:response) { File.expand_path("../fixtures/scishield/user_not_found.json", __dir__) }
-    before do
-      stub_request(:get, api_endpoint)
-        .to_return(
-          body: File.new(response),
-          status: 200,
-        )
+    describe "the user is not found" do
+      let(:response) { File.expand_path("../fixtures/scishield/user_not_found.json", __dir__) }
+      before do
+        stub_request(:get, api_endpoint)
+          .to_return(
+            body: File.new(response),
+            status: 200,
+          )
+      end
+
+      it "doesn't raise an error, and doesn't certify anything" do
+        random = instance_double(ResearchSafetyCertificate, name: "RANDOM")
+        expect(adapter).not_to be_certified(random)
+      end
     end
 
-    it "doesn't raise an error, and doesn't certify anything" do
-      random = instance_double(ResearchSafetyCertificate, name: "RANDOM")
-      expect(adapter).not_to be_certified(random)
+    describe "the user is found but the response includes an error" do
+      let(:response) { File.expand_path("../fixtures/scishield/errors.json", __dir__) }
+      before do
+        stub_request(:get, api_endpoint)
+          .to_return(
+            body: File.new(response),
+            status: 403,
+          )
+      end
+
+      it "raises an error" do
+        expect { adapter.certified?(anything) }.to raise_error(/Forbidden: This route can only be accessed by authenticated users/)
+      end
     end
   end
 
-  describe "the user is found but the response includes an error" do
-    let(:response) { File.expand_path("../fixtures/scishield/errors.json", __dir__) }
+  context "Scishield training exists in database" do
     before do
-      stub_request(:get, api_endpoint)
-        .to_return(
-          body: File.new(response),
-          status: 403,
-        )
+      3.times { create(:scishield_training, user_id: user.id) }
     end
 
-    it "raises an error" do
-      expect { adapter.certified?(anything) }.to raise_error(/Forbidden: This route can only be accessed by authenticated users/)
+    it "gets course names from database" do
+      expect(adapter.certified_course_names).to contain_exactly(
+        "Scishield Course Name 1",
```

Comment:
These course names depend on the factory - which can be a little brittle over time.   I'd vote for either setting the names explicitly in the spec, or using `let!` to create the trainings and then referencing that variable to get the expected names.